### DISCUSSION TITLE
Usage in raw HTML/Twig

### DISCUSSION THREAD

**Discussion by Nkzq** [ElMassimo/vite-plugin-image-presets#4] [Q&A] [answered]
Hi,

It looks like a great plugin ! Sorry if I misunderstand something, but I can't find how to use this in a raw HTML or a Twig template. Is this something possible ?

Best.

**Comment by ElMassimo** [comment, marked as answer]
Hi Romain!

The plugin relies on JS imports to detect the `preset` query parameter and generate images accordingly.

From your comment, I get that you are trying to use it from PHP? If that's the case, the plugin is not designed for that use case, although it does have a programmatic API that you could adapt to your purposes by using a Vite plugin.

Moving this to discussions, feel free to ask additional questions, or provide an example of how you would use it.

**Reply by Nkzq** [reply]
Thanks for your quick answer.

Aight I got it. I tried to use it in a Twig template with a simple <img> tag and hoping for the magic occurs on the HTML file generation 😁
